src/x86_64/hpet.c

```diff
@@ -117,22 +119,45 @@ static u64 hpet_main_counter()
 static void timer_config(int timer, timestamp rate, thunk t, boolean periodic)
 {
     if (!hpet_interrupts[timer]) {
-        u32 a, d;
+        u64 c = TCONF(32MODE_CNF) | TCONF(INT_ENB_CNF);
         hpet_interrupts[timer] = allocate_interrupt();
-        msi_format(&a, &d, hpet_interrupts[timer]);
-        hpet->timers[timer].fsb_int = ((u64)a << 32) | d;
+        if (hpet->timers[timer].config & TCONF(FSB_INT_DEL_CAP)) {
+            u32 a, d;
+            msi_format(&a, &d, hpet_interrupts[timer]);
+            hpet->timers[timer].fsb_int = ((u64)a << 32) | d;
+            c |= TCONF(FSB_EN_CNF);
+        } else {
+            u32 route_cap = (hpet->timers[timer].config >>
+                    HPET_TIMER_CONFIG_INT_ROUTE_CAP_SHIFT) &
+                    MASK(HPET_TIMER_CONFIG_INT_ROUTE_CAP_BITS);
+            int gsi = -1;
+            for (int i = 0; i < 64; i++) {
```

Comment:
Should this be i < 32? route_cap is a u32.

Reply:
Yes, I changed it to i < 32